src/drivers/net/intel.c: add ID for i219-LM rev 20

mkopec opened the pull request:
Add PCI device ID `550a` for i219lm-20 found in a Clevo V540TU (Meteor Lake chipset)

NiKiZe commented:
Thanks for your PR!

Are you sure this is correct? https://admin.pci-ids.ucw.cz/read/PC/8086/550a

There should be more variants, also please see existing commits and ipxe.org/admin for guidelines on commit headers and messages.

mkopec commented:
Hi, sorry for the late response.

The ID is also in Linux sources, but with a different rev: https://github.com/torvalds/linux/blob/55027e689933ba2e64f3d245fb1ff185b3e7fc81/drivers/net/ethernet/intel/e1000e/hw.h#L109

On actual hardware, lspci reports rev20, so I'm not sure which is more correct.

I haven't addressed the review yet because I'm working on resolving some bugs with this model on iPXE. I get this behavior where iPXE can get IP from DHCP if I've rebooted to iPXE from Linux, but it doesn't work when coldbooting from off (G3) straight to iPXE. It sees that the link is up but can't establish a connection. Some logs:

```
**iPXE initialising devices...
INTEL 0x67abba50 MAC reset (00180260/40080683 was 00180240/40080683)
INTEL 0x67abba50 has autoloaded MAC address d4:93:90:3e:0c:78
iPXE 1.21.1+ (gbaceb) -- Open Source Network Boot Firmware -- https://ipxe.org
Features: DNS HTTP HTTPS iSCSI TFTP VLAN SRP AoE EFI Menu

PiPXE> dhcp
INTEL 0x67abba50 WARNING: applying reset hang workaround
INTEL 0x67abba50 ring 03800 is at [67d19000,67d19100)
INTEL 0x67abba50 ring 02800 is at [67d13000,67d13100)
INTEL 0x67abba50 RX 0 is [67abc800,67abd000)
INTEL 0x67abba50 RX 1 is [67abe800,67abf000)
INTEL 0x67abba50 RX 2 is [67abf000,67abf800)
INTEL 0x67abba50 RX 3 is [67abf800,67ac0000)
INTEL 0x67abba50 RX 4 is [67ac0000,67ac0800)
INTEL 0x67abba50 RX 5 is [67ac0800,67ac1000)
INTEL 0x67abba50 RX 6 is [67ac1000,67ac1800)

INTEL 0x67abba50 RX 7 is [67ac1800,67ac2000)
INTEL 0x67abba50 link status is40080683
Configuring (net0 d4:93:90:3e:0c:78)...INTEL 0x67abba50 TX 0 is [67ac203e,67ac21
f3)
INTEL 0x67abba50 TX 0 complete
INTEL 0x67abba50 TX 1 is [67abd132,67abd178)
INTEL 0x67abba50 TX 1 complete
INTEL 0x67abba50 TX 2 is [67abd132,67abd144)
INTEL 0x67abba50 TX 2 complete
INTEL 0x67abba50 TX 3 is [67abd132,67abd178)
INTEL 0x67abba50 TX 3 complete
INTEL 0x67abba50 TX 4 is [67abd132,67abd178)
INTEL 0x67abba50 TX 4 complete
INTEL 0x67abba50 TX 5 is [67abd172,67abd178)
INTEL 0x67abba50 TX 5 complete
.INTEL 0x67abba50 TX 6 is [67ac203e,67ac21f3)
INTEL 0x67abba50 TX 6 complete
INTEL 0x67abba50 TX 7 is [67abd132,67abd178)
INTEL 0x67abba50 TX 7 complete
.INTEL 0x67abba50 TX 8 is [67abd132,67abd144)
INTEL 0x67abba50 TX 8 complete
.INTEL 0x67abba50 TX 9 is [67ac203e,67ac21f3)
INTEL 0x67abba50 TX 9 complete
.INTEL 0x67abba50 TX 10 is [67abc632,67abc644)
INTEL 0x67abba50 TX 10 complete
...INTEL 0x67abba50 TX 11 is![67ac203e,67ac21f3)
INTEL 0x67abba50 TX 11 complete
........ No configuration methods succeeded (https://ipxe.org/040ee186)
INTEL 0x67abba50 MAC reset (00180260/40080683 was 00180240/40080683)
iPXE>**
```

I'll try to update this PR if I find a solution

stappersg commented:
> I'll try to update this PR if I find a solution

Please do

NiKiZe commented:
So 8086:550a is clearly i219-lm (18) https://admin.pci-ids.ucw.cz/read/PC/8086/550a
Ignoring the name, we would expect the patch to work fine on the hardware, if this fails I would like to see `ifstat` output after `dhcp`
Any chance you still have access to hardware and can test?

Also if you can, I'm curious what you get from `lspci -nn | grep Ether`

NiKiZe commented:
The "rev 20" confusion seems to come from:
`00:1f.6 Ethernet controller [0200]: Intel Corporation Device [8086:550a] (rev 20)`
